test/integration/acceptance/create_domain_test.cpp

```diff
@@ -0,0 +1,217 @@
+  IntegrationTestFramework()
+      .setInitialState(kAdminKeypair)
+      .sendTx(makeUserWithPerms())
+      .skipProposal()
+      .skipBlock()
+      .sendTx(completeTx(baseTx().createDomain(kNewDomain, inexistent_role)))
+      .skipProposal()
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 0); })
+      .done();
+}
+
+/**
+ * @given some user with can_create_domain permission
+ * @when execute tx with CreateDomain command with already existing domain
+ * @then there is no tx in proposal
+ */
+TEST_F(CreateDomain, ExistentName) {
+  const std::string &existing_domain = IntegrationTestFramework::kDefaultDomain;
+  IntegrationTestFramework()
+      .setInitialState(kAdminKeypair)
+      .sendTx(makeUserWithPerms())
+      .skipProposal()
+      .skipBlock()
+      .sendTx(completeTx(baseTx().createDomain(existing_domain, kRole)))
+      .skipProposal()
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 0); })
+      .done();
+}
+
+/**
+ * @given some user with can_create_domain permission
+ * @when execute tx with CreateDomain command with maximum available length
+ * @then there is the tx in proposal
+ */
+TEST_F(CreateDomain, MaxLenName) {
+  IntegrationTestFramework()
+      .setInitialState(kAdminKeypair)
+      .sendTx(makeUserWithPerms())
+      .skipProposal()
+      .skipBlock()
+      .sendTx(completeTx(baseTx().createDomain(std::string(9, 'a'), kRole)))
+      .skipProposal()
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 1); })
+      .done();
+}
+
+/**
+ * @given some user with can_create_domain permission
+ * @when execute tx with CreateDomain command with too long length
+ * @then the tx hasn't passed stateless validation
+ *       (aka skipProposal throws)
+ */
+TEST_F(CreateDomain, TooLongName) {
+  IntegrationTestFramework itf;
+  itf.setInitialState(kAdminKeypair)
+      .sendTx(makeUserWithPerms())
+      .skipProposal()
+      .skipBlock()
+      .sendTx(completeTx(baseTx().createDomain(std::string(10, 'a'), kRole)));
+  ASSERT_ANY_THROW(itf.skipProposal());
+}
+
+/**
+ * @given some user with can_create_domain permission
+ * @when execute tx with CreateDomain command with empty domain name
+ * @then the tx hasn't passed stateless validation
+ *       (aka skipProposal throws)
+ */
+TEST_F(CreateDomain, EmptyName) {
+  const std::string &empty_name = "";
+  IntegrationTestFramework itf;
+  itf.setInitialState(kAdminKeypair)
+      .sendTx(makeUserWithPerms())
+      .skipProposal()
+      .skipBlock()
+      .sendTx(completeTx(baseTx().createDomain(empty_name, kRole)));
+  ASSERT_ANY_THROW(itf.skipProposal());
+}
+
+/**
+ * @given some user with can_create_domain permission
+ * @when execute tx with CreateDomain command with empty role name
+ * @then the tx hasn't passed stateless validation
+ *       (aka skipProposal throws)
+ */
+TEST_F(CreateDomain, DISABLED_EmptyRole) {
```

Comment:
Please rename EmptyRole to EmptyRoleName, also in other descriptions. EmptyRole can also relate to role having no permissions.